src/SFA.DAS.PR.Jobs/Functions/Notifications/SendNotificationsFunction.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Microsoft.Azure.Functions.Worker;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.Notifications.Messages.Commands;
+using SFA.DAS.PR.Data;
+using SFA.DAS.PR.Data.Common;
+using SFA.DAS.PR.Data.Entities;
+using SFA.DAS.PR.Data.Repositories;
+using SFA.DAS.PR.Jobs.Configuration;
+using SFA.DAS.PR.Jobs.Services;
+
+namespace SFA.DAS.PR.Jobs.Functions.Notifications;
+
+public class SendNotificationsFunction
+{
+    private readonly ILogger<SendNotificationsFunction> _logger;
+    private readonly NotificationsConfiguration _notificationsConfiguration;
+    private readonly IFunctionEndpoint _functionEndpoint;
+    private readonly IProviderRelationshipsDataContext _providerRelationshipsDataContext;
+    private readonly INotificationRepository _notificationRepository;
+    private readonly ITokenService _tokenService;
+
+    public SendNotificationsFunction(
+        ILogger<SendNotificationsFunction> logger,
+        IConfiguration configuration,
+        IFunctionEndpoint functionEndpoint,
+        IProviderRelationshipsDataContext providerRelationshipsDataContext,
+        INotificationRepository notificationRepository,
+        ITokenService tokenService
+    )
+    {
+        _logger = logger;
+        _functionEndpoint = functionEndpoint;
+        _providerRelationshipsDataContext = providerRelationshipsDataContext;
+        _notificationRepository = notificationRepository;
+        _tokenService = tokenService;
+
+        _notificationsConfiguration = new NotificationsConfiguration();
+        configuration.GetSection("ApplicationConfiguration:Notifications").Bind(_notificationsConfiguration);
+    }
+
+    [Function(nameof(SendNotificationsFunction))]
+    public async Task<int> Run([TimerTrigger("%SendNotificationsFunctionSchedule%", RunOnStartup = true)] TimerInfo timer, FunctionContext executionContext, CancellationToken cancellationToken)
+    {
+        _logger.LogInformation("{FunctionName} has been triggered.", nameof(SendNotificationsFunction));
+
+        List<Notification> notifications = await _notificationRepository.GetPendingNotifications(
+            _notificationsConfiguration.BatchSize, 
+            NotificationType.Provider, 
+            cancellationToken
+        );
+
+        if(notifications.Count == 0)
```

Comment:
There is no need to return anything, is there? I would rather just log the count anyway, so we can see the job is working.